Scripts/dea_datahandling.py

```diff
@@ -704,3 +714,202 @@ def paths_to_datetimeindex(paths, string_slice=(0, 10)):
     date_strings = [os.path.basename(i)[slice(*string_slice)]
                     for i in paths]
     return pd.to_datetime(date_strings)
+
+
+def calc_geomedian(ds,
+                   axis="time",
+                   max_value=None,
+                   min_value=None,
+                   num_threads=1,
+                   eps=1e-7,
+                   nocheck=True):
+    '''
+    Runs geomedian over a xarray.Dataset or xarray.DataArray.
+
+    Specify min_value and max_value if known for better performance,
+    especially if using dask.
+
+    Parameters
+    ----------
+    da : xarray.DataArray object
+    max_value : the maximum value that could be found in the array
+    min_value : the minimum value that could be found in the array
+    '''
+    da = (odc.algo.reshape_for_geomedian(ds, axis=axis) if
+          isinstance(ds, xr.Dataset) else ds)
+
+    if max_value is None:
+        max_value = da.max(skipna=True)
+    if min_value is None:
+        min_value = da.min(skipna=True)
```

Comment:
So this will be really slow for dask inputs, it will essentially load all the data three times, one for max one for min and then again for actual computation.

Reply:
also it is likely to be broken for `nodata=-999` case as it will pick `-999` as min values, as I do not think that `skipna` skips `nodata` values, only float NaNs

Reply:
Scaling is only performed to improve numeric stability it is not a prerequisite for correct operation, so I would just skip scaling and convert to float with `*1+0` if user hasn't supplied min/max.

Also from API perspective maybe easier if that was something like `value_range=(min_val, max_val)` or some better name.

Reply:
For DE Africa, its handy having the `max_value` and `min_value` parameters as the satellite products have different surface reflectance scalings. Are the SR scaling values identical between all products for DEA? I think they might be (0-10,000), in which case we could remove the calculations and hard code the min and max values.

Reply:
also checkout `int_geomedian`, I think that gives better performance as it delays conversion to float until very end, just before calling to Dale's code, so it has lower memory and bandwidth requirement.

Reply:
Have you observed significant difference of output quality when selecting "exact" scaling factors vs none? 3 times slower operation and 3 times bandwidth better be worth it.

Reply:
Besides we are not preventing that use case, just changing default behaviour to be less "surprising". We can just document the fact that scaling can help somewhat and if you know rough range of the data you can supply it and maybe get slightly better results out of it. And if not -- nothing bad.

With code as is, it is broken when given completely uniform input, as is will attempt `/0`